src/main/java/thewaypointers/trafficsimulator/JunctionLocationTestStarter.java

```diff
@@ -18,9 +13,10 @@
 
     public static void main(String[] args){
 
-        IStateProvider simulation = new JunctionTestProvider();
-        mainFrame=new MainFrame(simulation.getNextState(0));
-        TrafficSimulatorManager.setMainFrame(mainFrame);
-        TrafficSimulatorManager.setSimulation(simulation);
+        TrafficSimulatorManager trafficSimulatorManager = new TrafficSimulatorManager();
+        IStateProvider simulation = new RoadNetworkProvider();
```

Comment:
Yes, but the class is a starter for `JunctionTestProvider`. If you want to start the program with `RoadNetworkProvider`, make another starter class for that. It's super confusing for other people when they pull changes and suddenly the junction test starter doesn't start the junction test anymore.

Besides, the map in `JunctionTestProvider` is created in the same way as in other places, so it should work as well (the junction labels work well for example). Did you mean it's still an unsolved bug that clicking doesn't work on that?